### DISCUSSION TITLE
Linux Shared object

### DISCUSSION THREAD

**Discussion by Faru0** [BishopFox/sliver#1211] [Q&A] [answered]
Hello, 

I generated a linux shared object. I have attempted to initialize via _start. no dice. I have been able to get the shared object (dll) working in windows with rundll and StartW. l have attempted many different things but i have been unable to get .so to run. 

generate --os linux --format shared --mtls IPADDR --save ./path/obj.so

Is there a way to specify a function to hook  when i generate sliver shared object? So if i execute via LD_PRELOAD i can hook functions such  as time, strcmp, printf, etc ?

**Comment by moloch--** [comment, marked as answer]
For `LD_PRELOAD` use the `--run-at-load` flag: `generate --os linux --format shared --run-at-load`

**Reply by lesnuages** [reply]
This, use `--run-at-load`. At the moment there's no way to configure a function to hook.
